arbor/morph/stitch.cpp

```diff
@@ -0,0 +1,214 @@
+#include <memory>
+#include <unordered_map>
+#include <vector>
+
+#include <arbor/morph/stitch.hpp>
+#include <arbor/morph/morphexcept.hpp>
+#include <arbor/morph/locset.hpp>
+#include <arbor/morph/region.hpp>
+#include <arbor/util/optional.hpp>
+
+#include "util/ordered_forest.hpp"
+#include "util/maputil.hpp"
+#include "util/meta.hpp"
+#include "util/rangeutil.hpp"
+#include "util/transform.hpp"
+
+namespace arb {
+
+struct stitch_builder_impl {
+    struct stitch_segment {
+        double along_prox;
+        double along_dist;
+
+        mpoint prox;
+        mpoint dist;
+        int tag;
+        std::string stitch_id;
+        msize_t seg_id;
+    };
+
+    using forest_type = util::ordered_forest<stitch_segment>;
+
+    forest_type forest;
+    std::unordered_map<std::string, forest_type::iterator> id_to_node;
+    std::string last_id;
+
+    stitch_builder_impl() = default;
+    stitch_builder_impl(stitch_builder_impl&&) = default;
+    stitch_builder_impl(const stitch_builder_impl& other):
+        forest(other.forest),
+        last_id(other.last_id)
+    {
+        for (auto i = forest.preorder_begin(); i!=forest.preorder_end(); ++i) {
+            id_to_node.insert({i->stitch_id, i});
+        }
+    }
+
+    stitch_builder_impl& operator=(stitch_builder_impl&&) = default;
+    stitch_builder_impl& operator=(const stitch_builder_impl& other) {
+        if (this==&other) return *this;
+        return *this = stitch_builder_impl(other);
+    }
+
+    void add(mstitch f, const std::string& parent, double along, bool infer_prox = false) {
```

Comment:
`infer_prox` is not actually used anywhere

Reply:
Nice catch! Yep, this is leftover from an earlier approach.